context/response/responses.go

```diff
@@ -0,0 +1,53 @@
+package response
+
+import (
+	beecontext "github.com/astaxie/beego/context"
+)
+
+// JSON renders value to the response as JSON
+func JSON(value interface{}, encoding ...bool) Renderer {
```

Comment:
Also, response.JSON can not be put on Output because Output already has JSON. and repsonse JSON again returns a renderer which eventually calls Output.JSON. 
The point of renderers what that writing to output is always done after the function is finished and not within the code.